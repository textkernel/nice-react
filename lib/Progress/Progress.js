import React from 'react';
import PropTypes from 'prop-types';
import props from '../props';

const Progress = ({ percentage, message, context }) => {
  const label = message !== '' ? message : `${percentage}%`;

  return ( 
    <div className={`nice-progress ${context}`}>
      <div className='nice-progress-fill' style={{ width: `${percentage}%` }} >
        {label}
      </div>
    </div>
  );
};

Progress.propTypes = {
  percentage: PropTypes.number.isRequired,
  message: PropTypes.string,
  context: PropTypes.oneOf(props.contexts)
};

Progress.defaultProps = {
  message: '',
  context: 'primary'
};

export default Progress;

