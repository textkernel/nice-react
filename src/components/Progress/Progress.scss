@import '../../styles/nice';

.nice-progress {
    background-color: darken($gray-light, 5%);
    height: 28px;
    padding: 4px;
    margin: 5px 0;
    @include nice-border-radius(15px);

    .nice-progress-fill {
        text-align: center;
        color: $color-white;
        background-color: $color-primary;
        max-width: 100%;
        min-width: 10%;
        text-overflow: ellipsis;
        overflow: hidden;
        @include nice-font($size: .8em, $weight: 600);
        @include nice-line-height(20px);
        @include nice-border-radius(15px);
        @include nice-transition(width 0.3s ease-in-out);
        @include nice-text-shadow(1px 1px 0 rgba(0,0,0,.5));
    }
    @each $type in (brand, primary, accent, info, good, warning, bad) {
        &.#{$type} {
            .nice-progress-fill {
                background-color: map-get($color-scheme, $type);
            }
        }
    }
    &.pulse {
        .nice-progress-fill {
            @include nice-animation(pulse 1.5s ease-out 0s infinite);
        }
    }
}
