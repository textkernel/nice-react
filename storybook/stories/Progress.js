import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, text, boolean, select, number } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import props from '../../src/props';

import { Progress } from 'nice-react';

const stories = storiesOf('Progress', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('with zero percent', () => (
    <Progress percentage={0} />
));

stories.add('with context', () => (
    <Progress percentage={80} context="accent" />
));

stories.add('with custom message', () => (
    <Progress percentage={80} context="good" message="Loading..." />
));

stories.add('with dynamic properties', () => {
    const contextOptions = props.contexts;
    const context = select('Context', contextOptions, props.contexts[0]);
    const message = text('Message', '');
    const percentage = number('Percentage', 50);

    return <Progress percentage={percentage} context={context} message={message} />
});
