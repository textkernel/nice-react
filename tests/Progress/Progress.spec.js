import React from 'react';
import Progress from '../../lib/Progress';

describe('<Progress />', () => {
  it('should render the Progress', () => {
    const wrapper = shallow(<Progress percentage={50} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Progress with context', () => {
    const wrapper = shallow(<Progress percentage={50} context='good' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Progress with custom message', () => {
    const wrapper = shallow(<Progress percentage={50} message='Loading...' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Progress with custom message and context', () => {
    const wrapper = shallow(<Progress percentage={50} message='Loading...' context='bad' />);
    expect(wrapper).toMatchSnapshot();
  });
});