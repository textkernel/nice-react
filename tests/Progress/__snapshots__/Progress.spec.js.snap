// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should render the Progress 1`] = `
<div
  className="nice-progress primary"
  style={Object {}}
>
  <div
    className="nice-progress-fill"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`<Progress /> should render the Progress with context 1`] = `
<div
  className="nice-progress good"
  style={Object {}}
>
  <div
    className="nice-progress-fill"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`<Progress /> should render the Progress with custom className/style 1`] = `
<div
  className="nice-progress primary my-custom-class"
  style={
    Object {
      "top": 0,
    }
  }
>
  <div
    className="nice-progress-fill"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    0%
  </div>
</div>
`;

exports[`<Progress /> should render the Progress with custom message 1`] = `
<div
  className="nice-progress primary"
  style={Object {}}
>
  <div
    className="nice-progress-fill"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    Loading...
  </div>
</div>
`;

exports[`<Progress /> should render the Progress with custom message and context 1`] = `
<div
  className="nice-progress bad"
  style={Object {}}
>
  <div
    className="nice-progress-fill"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    Loading...
  </div>
</div>
`;
